pkg/sources/postgres/dogs_test.go

```diff
@@ -113,7 +113,8 @@ func (s StoreTestSuite) TestUpdateDog() {
 		s.NoError(err)
 		createdDog.Name = "Lolita"
 
-		actualDog, err := s.store.UpdateDog(ctx, &dog)
+		s.NotEqual(dog.Name, createdDog.Name)
```

Comment:
As previously written, this test was actually enforcing the problematic aspects of the references to the same memory address.

If the previous version of the test had this new `NotEqual` line, it would have shown that `createdDog` and `dog` were referring to the same memory address, because only `createdDog.Name` was actually expected to have been updated.